File: server/src/main/java/ua/com/alevel/service/educationalprocess/TopicService.java
package ua.com.alevel.service.educationalprocess;

import org.springframework.stereotype.Service;
import ua.com.alevel.persistence.crudhelper.CrudRepositoryHelper;
import ua.com.alevel.persistence.entity.educationalprocess.Topic;
import ua.com.alevel.persistence.repository.educationalprocess.TopicRepository;
import ua.com.alevel.service.AbstractService;

@Service
public class TopicService extends AbstractService<Topic, TopicRepository> {

    private final TopicRepository topicRepository;
    private final CrudRepositoryHelper<Topic, TopicRepository> crudRepositoryHelper;

    public TopicService(TopicRepository topicRepository, CrudRepositoryHelper<Topic, TopicRepository> crudRepositoryHelper) {
        super(topicRepository, crudRepositoryHelper);
        this.topicRepository = topicRepository;
        this.crudRepositoryHelper = crudRepositoryHelper;
    }
}

File: client-tutor/src/app/shared/services/educationalprocess/topic.service.ts
import {Injectable} from '@angular/core';
import {AbstractService} from "../abstract.service";
import {HttpClient} from "@angular/common/http";
import {Topic} from "../../models/educationalprocess/topic.model";

@Injectable({
  providedIn: 'root'
})
export class TopicService extends AbstractService<Topic> {

  override resourceUrl = '/api/topics';

  constructor(public override httpClient: HttpClient) {
    super(httpClient);
  }
}
